src/backend/distributed/executor/local_executor.c

```diff
@@ -245,6 +246,101 @@ ExtractParametersForLocalExecution(ParamListInfo paramListInfo, Oid **parameterT
 }
 
 
+/*
+ * ExecuteLocalUtilityTaskList executes a list of tasks locally. This function
+ * also logs local execution notice for each task and sets
+ * TransactionAccessedLocalPlacement to true for next set of possible queries
+ * & commands within the current transaction block. See the comment in function.
+ */
+void
+ExecuteLocalUtilityTaskList(List *localTaskList)
+{
+	Task *localTask = NULL;
+
+	foreach_ptr(localTask, localTaskList)
+	{
+		const char *localTaskQueryCommand = TaskQueryString(localTask);
+
+		if (localTask->anchorShardId != INVALID_SHARD_ID)
+		{
+			/*
+			 * If anchor shard id is valid, then we really access a local
+			 * placement. We should register it to force the local execution
+			 * of the following commands withing the current transaction.
+			 * It can be a coordinated transaction or a transaction that is
+			 * initiated explicitly.
+			 */
+			TransactionAccessedLocalPlacement = true;
+		}
+
+		LogLocalCommand(localTask);
+
```

Comment:
maybe add `Assert (TaskAccessesLocalNode(localTask))` in case any code-path somehow triggers erroneously this